core/templates/dev/head/components/SearchBarDirective.js

```diff
@@ -100,43 +100,102 @@ oppia.directive('searchBar', [function() {
             selections[optionName] = !selections[optionName];
           }
 
-          _updateSelectionDetails(itemsType);
-          _onSearchQueryChangeExec();
+          updateSelectionDetails(itemsType);
+          onSearchQueryChangeExec();
         };
 
         $scope.deselectAll = function(itemsType) {
           $scope.selectionDetails[itemsType].selections = {};
-          _updateSelectionDetails(itemsType);
+          updateSelectionDetails(itemsType);
         };
 
-        var _searchBarFullyLoaded = false;
+        var searchBarFullyLoaded = false;
 
-        var _hasChangedSearchQuery = Boolean(urlService.getUrlParams().q);
-        var _onSearchQueryChangeExec = function() {
+        var hasChangedSearchQuery = Boolean(urlService.getUrlParams().q);
+        var onSearchQueryChangeExec = function() {
+          $scope.searchIsLoading = true;
           searchService.executeSearchQuery(
               $scope.searchQuery, $scope.selectionDetails.categories.selections,
               $scope.selectionDetails.languageCodes.selections, function() {
-            if (!_hasChangedSearchQuery && _searchBarFullyLoaded) {
-              _hasChangedSearchQuery = true;
+            if (!hasChangedSearchQuery && searchBarFullyLoaded) {
+              hasChangedSearchQuery = true;
               $rootScope.$broadcast('hasChangedSearchQuery');
             }
           });
+
+          var searchUrlQueryString = searchService.getSearchUrlQueryString(
+            $scope.searchQuery, $scope.selectionDetails.categories.selections,
+            $scope.selectionDetails.languageCodes.selections
+          );
+          if ($window.location.pathname == '/search/find') {
+            $location.url('/find?q=' + searchUrlQueryString);
+          } else if ($scope.searchQuery != '') {
+            $window.location.href = '/search/find?q=' + searchUrlQueryString;
+          }
         };
 
         // Initialize the selection descriptions and summaries.
         for (var itemsType in $scope.selectionDetails) {
-          _updateSelectionDetails(itemsType);
+          updateSelectionDetails(itemsType);
         }
 
         $scope.onSearchQueryChange = function(evt) {
           // Query immediately when the enter or space key is pressed.
           if (evt.keyCode == 13 || evt.keyCode == 32) {
-            _onSearchQueryChangeExec();
+            onSearchQueryChangeExec();
           } else {
-            oppiaDebouncer.debounce(_onSearchQueryChangeExec, 400)();
+            oppiaDebouncer.debounce(onSearchQueryChangeExec, 400)();
+          }
+        };
+
+        var updateSearchFields = function(itemsType, urlComponent, pattern,
+                                          smallerPatt) {
+          var itemCodes = urlComponent.match(pattern);
+
+          var items = [];
+          if (itemCodes != undefined) {
+            items = itemCodes[0].match(pattern);
```

Comment:
The first match is to find all the items with quotes and the second match is to find items in between those quotes. I tried doing parenthesis around the letters like `/"([a-z]{2})"/g` but I'm still getting `"en"` instead of `en`.